pkg/cmd/util/flags/flags.go

```diff
@@ -33,3 +35,96 @@ func Resolve(args map[string][]string, fn func(*pflag.FlagSet)) []error {
 	fn(fs)
 	return Apply(args, fs)
 }
+
+// ComponentFlag represents a set of enabled components used in a command line.
+type ComponentFlag struct {
+	enabled  string
+	disabled string
+	isSet    func() bool
+
+	calculated sets.String
+
+	allowed  sets.String
+	mappings map[string][]string
+}
+
+// NewComponentFlag returns a flag that represents the allowed components and can be bound to command line flags.
+func NewComponentFlag(mappings map[string][]string, allowed ...string) *ComponentFlag {
+	set := sets.NewString(allowed...)
+	return &ComponentFlag{
+		allowed:  set,
+		mappings: mappings,
+		enabled:  strings.Join(set.List(), ","),
+		isSet:    func() bool { return false },
+	}
+}
+
+// Disable marks the provided components as disabled.
+func (f *ComponentFlag) Disable(components ...string) {
```

Comment:
same here, this interacts oddly depending on whether it's called before or after args are parsed